overlord/snapstate/handlers.go

```diff
@@ -3252,3 +3279,26 @@ func InjectAutoConnect(mainTask *state.Task, snapsup *SnapSetup) {
 	InjectTasks(mainTask, state.NewTaskSet(autoConnect))
 	mainTask.Logf("added auto-connect task")
 }
+
+// GetSnapDirOptions returns *dirs.SnapDirOptions configured according to the
+// enabled experimental features. The state must be locked by the caller.
+var GetSnapDirOptions = getSnapDirOptions
+
+func getSnapDirOptions(state *state.State) (*dirs.SnapDirOptions, error) {
```

Comment:
not for here but this will need to take a snap name in the world where we migrate a snap at a time